src/gluonts/dataset/split/splitter.py

```diff
@@ -284,28 +335,24 @@ class OffsetSplitter(pydantic.BaseModel, AbstractBaseSplitter):
         of excluded values is enough for the test case, i.e., at least
         ``prediction_length`` (for ``rolling_split`` multiple of
         ``prediction_length``) values are left off.
-    max_history
-        If given, all entries in the *test*-set have a max-length of
-        `max_history`. This can be used to produce smaller file-sizes.
     """
 
-    prediction_length: int
     split_offset: int
-    max_history: Optional[int] = None
 
     def _train_slice(self, item: TimeSeriesSlice) -> TimeSeriesSlice:
         return item[: self.split_offset]
 
     def _test_slice(
-        self, item: TimeSeriesSlice, offset: int = 0
-    ) -> TimeSeriesSlice:
-        offset_ = self.split_offset + offset + self.prediction_length
+        self, item: TimeSeriesSlice, prediction_length: int, offset: int = 0
+    ) -> Tuple[TimeSeriesSlice, TimeSeriesSlice]:
+        offset_ = self.split_offset + offset + prediction_length
         if self.split_offset < 0 and offset_ >= 0:
             offset_ += len(item)
-        return item[:offset_]
+        return (item[: self.split_offset], item[self.split_offset : offset_])
```

Comment:
Then this should probably be
```suggestion
        return (item[: offset_], item[offset_: offset_ + prediction_length])
```